File: breathing-animation/cli/lua/breathing.lua
-- Mindful Breathing Visualizer - Lua CLI
-- SWEBOK v4 Serene Palette: Emerald, Blue, Rose

local SERENE_EMERALD = "\27[38;2;52;211;153m"
local SERENE_BLUE    = "\27[38;2;96;165;250m"
local SERENE_ROSE    = "\27[38;2;251;113;133m"
local RESET          = "\27[0m"
local BEEP           = "\7"

-- Box Breathing Phases (4s In, 4s Hold, 4s Out, 4s Hold)
local phases = {
    { name = "Inhale", duration = 4, color = SERENE_EMERALD },
    { name = "Hold",   duration = 4, color = SERENE_BLUE },
    { name = "Exhale", duration = 4, color = SERENE_ROSE },
    { name = "Hold",   duration = 4, color = SERENE_BLUE }
}

-- Utility: Sleep function
local function sleep(seconds)
    os.execute("sleep " .. tonumber(seconds))
end

-- Utility: Clear Screen
local function clear_screen()
    os.execute("clear")
end

print("Starting Mindful Breathing (Lua)...")
sleep(1)

-- Main Loop
while true do
    for _, phase in ipairs(phases) do
        clear_screen()
        
        -- Audio Feedback (Beep at start of phase)
        io.write(BEEP)
        io.flush()
        
        -- Display Phase
        print(phase.color .. "--------------------------------")
        print("          " .. string.upper(phase.name))
        print("--------------------------------" .. RESET)
        
        -- Countdown
        for i = 1, phase.duration do
            io.write(phase.color .. string.rep("=", i * 2) .. "> " .. RESET)
            io.flush()
            sleep(1)
        end
        print("\n")
    end
end

File: breathing-animation/cli/perl/breathing.pl
#!/usr/bin/perl
# Mindful Breathing Visualizer - Perl CLI
# SWEBOK v4 Serene Palette

use strict;
use warnings;
use Time::HiRes qw(sleep);

# Define Palette (ANSI TrueColor)
my $SERENE_EMERALD = "\e[38;2;52;211;153m";
my $SERENE_BLUE    = "\e[38;2;96;165;250m";
my $SERENE_ROSE    = "\e[38;2;251;113;133m";
my $RESET          = "\e[0m";
my $BEEP           = "\7";

# Box Breathing Phases
my @phases = (
    { name => "Inhale", duration => 4, color => $SERENE_EMERALD },
    { name => "Hold",   duration => 4, color => $SERENE_BLUE },
    { name => "Exhale", duration => 4, color => $SERENE_ROSE },
    { name => "Hold",   duration => 4, color => $SERENE_BLUE }
);

print "Starting Mindful Breathing (Perl)...\n";
sleep(1);

# Main Loop
while (1) {
    foreach my $phase (@phases) {
        # Clear Screen
        print "\e[2J\e[H";
        
        # Audio Feedback
        print $BEEP;
        
        # Display Header
        print $phase->{color} . "--------------------------------\n";
        print "          " . uc($phase->{name}) . "\n";
        print "--------------------------------" . $RESET . "\n";
        
        # Countdown
        for (my $i = 1; $i <= $phase->{duration}; $i++) {
            my $bar = "=" x ($i * 2);
            print $phase->{color} . $bar . "> " . $RESET . "\r";
            $| = 1; # Flush buffer
            sleep(1);
        }
        print "\n";
    }
}
